### DISCUSSION TITLE
Can DAFoam ( mphys"scenario_aerostructural") support one wall patch for aerostruct opt, another patch for aerodynamic opt simultaneously?

### DISCUSSION THREAD

**Discussion by NUAA-qinfei** [mdolab/dafoam#701] [General]
I have two wall patch, "wing" and "nacelle", and have two FFD blocks,one FFD controls the "wing"shape, another FFD controls the "nacelle" position, and the "wing" patch have struct mesh , but "nacelle" patch no struct, I konw for the "wing" opt should use "scenario_aerostructural" ,but for the "nacelle" just the "scenario_areodynamic",I want to know if DAFoam can support this opt simultaneously,or "scenario_aerostructural" support pure aerodynamic opt  ? I do this but always have error. (if no wing struct,I do it successfully,just like the picture show,but now,I want to couple the wing struct to consider the wing static aeroelasticity)
![image](https://github.com/user-attachments/assets/94b3d6db-8372-48a6-9dea-457f9800ea48)
![image](https://github.com/user-attachments/assets/0c6f585d-e55a-4fb9-a238-7bbdce46fe9d)
![image](https://github.com/user-attachments/assets/de92e628-1ee3-4710-bd8a-79b7d0dcb853)
![image](https://github.com/user-attachments/assets/c4b21314-05a9-4df1-9889-f4475b0e4105)

**Comment by friedenhe** [comment]
As I understand, you are trying to do an aerostructure optimization for a full aircraft configuration. You generate CFD meshes for the entire aircraft, but you have FEM (structural) mesh for the wing only. This should be doable. What you need to do is to use `["wing", "nacelle"]` for the "designSurfaces" key in daOption, and use  `"couplingSurfaceGroups": {"wingGroup": ["wing"] }` for the "couplingInfo" key in daOption. Note that the aero-structural coupling patches can be different from the designSurface patches.

**Reply by NUAA-qinfei** [reply]
Thanks for prof,I do it as you say ,but always have error,and how to understand " the aero-structural coupling patches can be different from the designSurface patches.",why  different?

**Reply by NUAA-qinfei** [reply]
![Screenshot from 2024-11-21 09-39-50](https://github.com/user-attachments/assets/fcf3c324-84f7-4a28-b0e4-667b2453b5ee)
I do it as you say,always have this error after reading the bdf,

**Comment by friedenhe** [comment]
The above should work.. If not, maybe there is a bug somewhere in the code.. A temporal workaround is to use `["wing", "nacelle"]` for both the designSurfaces and "couplingSurfaceGroups"-"wingGroup".  The load transfer component uses the nearest N points (controlled by `n=xx` in `MeldBuilder`) to interpolate the force, so if your nacelle is far from the wing, its force should not impact the wing loading by a lot. Again, this is not a perfect solution, but it should get the case running.
